src/mrack/providers/aws.py

```diff
@@ -101,20 +98,20 @@ async def create_server(self, req):
             MinCount=1,
             MaxCount=1,
             InstanceType=specs.get("flavor"),
+            KeyName=self.ssh_key,
+            SecurityGroupIds=[self.sec_group],
         )
 
         ids = [srv.id for srv in aws_res]
-        assert len(ids) == 1  # ids must be len of 1 as we provision one vm at the time
-
+        if len(ids) != 1:  # ids must be len of 1 as we provision one vm at the time
+            raise (ProvisioningError)
```

Comment:
```python
raise ProvisioningError("Unexpected number of instances provisioned.")
```
_That is just a suggestion, feel free to improve the message._

Reply:
Sure!